tf2onnx/optimizer/merge_duplicated_nodes.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+"""Merge Duplicated Nodes.
+   Remove duplicate nodes except identity nodes which should be handled by identity optimizer
```

Comment:
add examples for this?

Reply:
the reason I change "optimize" to return graph is to avoid forcing user to modify the graph in place, sometime they can decide to copy a graph and modify it and return it.

Reply:
output_names is used for consistency to previous design......

Reply:
who needs that now? maybe we can do that later. Graph deep copy requires some changes in Graph/Node I think? it's not a simple call I guess.